internal/models/sessions.go

```diff
@@ -104,6 +106,54 @@ func (s *Session) UpdateOnlyRefreshInfo(tx *storage.Connection) error {
 	return tx.UpdateOnly(s, "refreshed_at", "user_agent", "ip")
 }
 
+type SessionValidityReason = int
+
+const (
+	SessionValid        SessionValidityReason = iota
+	SessionPastNotAfter                       = iota
+	SessionPastTimebox                        = iota
+	SessionTimedOut                           = iota
+)
+
+func (s *Session) CheckValidity(now time.Time, refreshTokenTime *time.Time, timebox, inactivityTimeout *time.Duration) SessionValidityReason {
```

Comment:
would be easier to read if arranged like
```suggestion
func (s *Session) CheckValidity(timebox, inactivityTimeout *time.Duration, now time.Time, refreshTokenTime *time.Time, ) SessionValidityReason {
```

Reply:
we are also using a mix of `*time.Time`  and `time.Time` here - can we pick one and be consistent?

Reply:
`*time.Time` means it can be nullable. `now` is integral and it can't be nullable. It is consistent.

Reply:
IDK to me setting `now` is more important than setting the nullable ones, so that's why it's first.